### PR TITLE
pythonPackages.python-vipaccess: init at 0.10.3

### PR DESCRIPTION
**Pull request by herrwiese**
###### Motivation for this change

`python-vipaccess` is a free and open source software (FOSS) implementation of Symantec's VIP Access client.  This, for example, allows provisioning TOTP on PayPal.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/python-vipaccess/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  };
+
+  propagatedBuildInputs = [
+    lxml
+    oath
+    pycryptodome
+    requests
+  ];
+
+  doCheck = false;
```
Comment (FRidh) [inline pkgs/development/python-modules/python-vipaccess/default.nix]: Why are the tests disabled? Include a comment in the expression explaining why the tests are disabled.

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/oath/default.nix [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1xqgcqgx6aa0j21hwsdb3aqpqhviwj756bcqjjjcm1h1aij11p6m";
+  };
+
+  meta = {
+    description = "Python implementation of the three main OATH specifications: HOTP, TOTP and OCRA";
+    homepage = "https://github.com/bdauvergne/python-oath";
+    license = stdenv.lib.licenses.bsd3;
```
Comment (worldofpeace) [inline pkgs/development/python-modules/oath/default.nix]:
```suggestion
    license = licenses.bsd3;
```

File: pkgs/development/python-modules/oath/default.nix [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    maintainers = with stdenv.lib.maintainers; [ aw ];
```
Comment (worldofpeace) [inline pkgs/development/python-modules/oath/default.nix]:
```suggestion
    maintainers = with maintainers; [ aw ];
```

File: pkgs/development/python-modules/oath/default.nix [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+buildPythonPackage rec {
+  pname = "oath";
+  version = "1.4.3";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1xqgcqgx6aa0j21hwsdb3aqpqhviwj756bcqjjjcm1h1aij11p6m";
+  };
+
+  meta = {
```
Comment (worldofpeace) [inline pkgs/development/python-modules/oath/default.nix]:
```suggestion
  meta = with stdenv.lib; {
```

File: pkgs/development/python-modules/python-vipaccess/default.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+  meta = {
+    description = "A free software implementation of Symantec's VIP Access application and protocol";
+    homepage = "https://github.com/dlenski/python-vipaccess";
+    license = stdenv.lib.licenses.asl20;
```
Comment (worldofpeace) [inline pkgs/development/python-modules/python-vipaccess/default.nix]:
```suggestion
    license = licenses.asl20;
```

File: pkgs/development/python-modules/python-vipaccess/default.nix [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    maintainers = with stdenv.lib.maintainers; [ aw ];
```
Comment (worldofpeace) [inline pkgs/development/python-modules/python-vipaccess/default.nix]:
```suggestion
    maintainers = with maintainers; [ aw ];
```

File: pkgs/development/python-modules/python-vipaccess/default.nix [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+  propagatedBuildInputs = [
+    lxml
+    oath
+    pycryptodome
+    requests
+  ];
+
+  doCheck = false;
+
+  meta = {
```
Comment (worldofpeace) [inline pkgs/development/python-modules/python-vipaccess/default.nix]:
```suggestion
  meta = with stdenv.lib; {
```

**Review by worldofpeace — CHANGES_REQUESTED** [review]
Verified that `oath` tests succeed and builds appropriately.

Following comments are just expression style.

File: pkgs/development/python-modules/python-vipaccess/default.nix [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+  propagatedBuildInputs = [
+    lxml
+    oath
+    pycryptodome
+    requests
+  ];
+
+  # requires a network connection
+  doCheck = false;
```
Comment (risicle) [inline pkgs/development/python-modules/python-vipaccess/default.nix]:
```
  checkInputs = [ pytest ];
  checkPhase = ''
    mv vipaccess vipaccess.hidden
    pytest tests/ -k 'not (test_check_token_detects_valid_totp_token
      or test_check_token_detects_valid_hotp_token
      or test_check_token_detects_invalid_token)'
  '';
```

Comment (herrwiese) [reply pkgs/development/python-modules/python-vipaccess/default.nix]: Ah, thanks, I didn't know how to do that.  I shortened the `-k` argument to `-k not test_check_token`, though. Works on NixOS.

Comment (risicle) [reply pkgs/development/python-modules/python-vipaccess/default.nix]: Yes I forget it's a pattern match.

**Comment by worldofpeace** [comment]
Made a report upstream about fixing up the tests https://github.com/dlenski/python-vipaccess/issues/30.

**Review by worldofpeace — APPROVED** [review]
LGTM :+1:

**Pull request merged**